genshin/client/components/hoyolab.py

```diff
@@ -113,18 +113,32 @@ async def redeem_code(
         code: str,
         uid: typing.Optional[int] = None,
         *,
+        game: typing.Optional[types.Game] = None,
         lang: typing.Optional[str] = None,
     ) -> None:
-        """Redeems a gift code for the current genshin user."""
-        uid = uid or await self._get_uid(types.Game.GENSHIN)
+        """Redeems a gift code for the current user."""
+        if game is None:
+            if self.default_game is None:
+                raise RuntimeError("No default game set.")
+
+            game = self.default_game
+
+        if game == types.Game.GENSHIN:
```

Comment:
Wanted to call out this section here. Currently there is no logic involving utilization of game_biz that requires detection of the current game other than here, but this may be something that should be factored out as its own utility function? If so, where would you like it?

Reply:
You mean like `_get_uid` functions now?

Reply:
(huh, those were not the lines I selected...)

I mean, `_get_uid` involves caching whereas a game maps 1:1 with a game_biz value. It'd either be a simple dict or a function that acts like one. A lookup.

Reply:
Oh in that case it should go in `utility/uid.py` ig.

Reply:
I think it's many-to-one no? Game_biz also determines whether it's the beta server and such.

Reply:
Oh I suppose you're right...crap. But the client only supports prod?

Reply:
`get_accounts` may return beta accounts too. No need to support them directly though.

Reply:
I've renamed the function to clarify that the values are prod-only. Seeing as they were only being used in a function that expected you to be on prod, utilization/expectations don't change. Should be good?

Reply:
yeah this is good.